rocksdb_admin/helix_client.cpp

```diff
@@ -109,8 +114,12 @@ void invokeClass(JNIEnv* env,
   env->SetObjectArrayElement(args, 12, env->NewStringUTF("--configPostUrl"));
   env->SetObjectArrayElement(args, 13,
                              env->NewStringUTF(config_post_url.c_str()));
+  if (FLAGS_use_s3_backup) {
+    env->SetObjectArrayElement(args, 14, env->NewStringUTF("--s3Bucket"));
+    env->SetObjectArrayElement(args, 15, env->NewStringUTF(FLAGS_s3_bucket.c_str()));
+  }
   if (disable_spectator) {
-    env->SetObjectArrayElement(args, 14, env->NewStringUTF("--disableSpectator"));
+    env->SetObjectArrayElement(args, FLAGS_use_s3_backup ? 16 : 14, env->NewStringUTF("--disableSpectator"));
```

Comment:
what is the logic here? why are `disableSpectator` and `use_s3_backup` related?

Reply:
they are not related. but we need to calculate the number of arguments passed to the jni